File: numberguess.java
import java.util.Scanner;

public class numberguess {
    public static void main(String[] args) {

        final int min = 1;
        final int max = 100;

        final int number = (int) (Math.random() * max - min + 1) + min;

        int tries = 1;

        System.out.println("\nGuess a number between " + min + " and " + max + ".");

        Scanner Input = new Scanner(System.in);

        while (true) {

            System.out.print("\nEnter your guess: ");
            int guess = Input.nextInt();

            if (guess == number) {
                System.out.println("\nYour guess of " + guess + " was correct!");
                System.out.println("It took you " + tries + " tries.");
                break;
            } else if (guess > number) {
                System.out.println("Your guess of " + guess + " was too high...");
                tries++;
                continue;
            } else if (guess < number) {
                System.out.println("Your guess of " + guess + " was too low...");
                tries++;
                continue;
            }
        }
        Input.close();
    }
}

File: Number Guess/NumberGuess.cpp
#include "NumberGuess.h"
#include <iostream>
#include <ctime>

int NumberGuess::generate_number(int min, int max)
{
	srand(time(NULL));
	int number = rand() % max + min;
	return number;
}

void NumberGuess::guess_number(int number)
{
	int tries = 1;

	while (true) {
		int guess;
		std::cout << "\nEnter your guess: ";
		std::cin >> guess;

		if (guess == number) {
			std::cout << "Correct, the number was " << number << ". It took you " << tries << " tries!" << std::endl;
			break;
		}

		else if (guess > number)
		{
			std::cout << "Incorrect, your guess was too high..." << std::endl;
			tries++;
			continue;
		}

		else if (guess < number)
		{
			std::cout << "Incorrect, your guess was too low..." << std::endl;
			tries++;
			continue;
		}
	}
}